attestation-agent/attester/src/tpm/mod.rs

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2025 Confidential Containers Project Authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use super::{Attester, InitDataResult, TeeEvidence};
+use crate::tpm_utils::{
+    detect_tpm_device, extend_pcr, generate_rsa_ak, get_quote, read_all_pcrs, TpmQuote,
+};
+use anyhow::*;
+use base64::Engine;
+use log::info;
+use serde::{Deserialize, Serialize};
+use tss_esapi::traits::Marshall;
+
+/// Evidence structure for the TPM Attester.
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Evidence {
+    pub svn: String,
+    pub report_data: String,
+    pub tpm_quote: TpmQuote,
+    pub ak_public: String,
+}
+
+#[derive(Debug, Default)]
+pub struct TpmAttester;
+
+pub fn detect_platform() -> bool {
+    // Return true if TPM device is detected
+    detect_tpm_device().is_some()
+}
+
+#[async_trait::async_trait]
+impl Attester for TpmAttester {
+    /// Get evidence for the TPM attester.
+    async fn get_evidence(&self, report_data: Vec<u8>) -> Result<TeeEvidence> {
+        let tpm_device = detect_tpm_device().ok_or_else(|| anyhow!("No TPM device found"))?;
+        info!("[TPM Attester] Using TPM device: {}", tpm_device);
+        std::env::set_var("TCTI", format!("device:{}", tpm_device));
```

Comment:
Can we avoid setting that env and use a field in the TpmAttester struct instead? (set_env is [marked unsafe](https://doc.rust-lang.org/std/env/fn.set_var.html) in more recent rust releases)